src/GovUk.Education.ExploreEducationStatistics.Content.Services/ReleaseService.cs

```diff
@@ -24,37 +26,76 @@ public class ReleaseService : IReleaseService
         private readonly IFileStorageService _fileStorageService;
         private readonly IMethodologyService _methodologyService;
         private readonly IUserService _userService;
+        private readonly IPublicationService _publicationService;
+        private readonly IMapper _mapper;
 
         public ReleaseService(
             IPersistenceHelper<ContentDbContext> persistenceHelper,
             IFileStorageService fileStorageService,
             IMethodologyService methodologyService,
-            IUserService userService)
+            IUserService userService,
+            IPublicationService publicationService,
+            IMapper mapper)
+
         {
             _persistenceHelper = persistenceHelper;
             _fileStorageService = fileStorageService;
             _methodologyService = methodologyService;
             _userService = userService;
+            _publicationService = publicationService;
+            _mapper = mapper;
         }
 
-        public async Task<Either<ActionResult, ReleaseViewModel>> Get(string publicationPath, string releasePath)
+        public async Task<Either<ActionResult, ReleaseViewModel>> Get(string publicationSlug, string? releaseSlug = null)
         {
-            return await CreateFromCachedPublicationAndRelease(publicationPath, releasePath,
-                    (publication, release) => new ReleaseViewModel(release, publication))
-                .OnSuccessCombineWith(model => _methodologyService.GetSummariesByPublication(model.Publication.Id))
-                .OnSuccess(viewModelAndMethodologies =>
+            return await _publicationService.Get(publicationSlug)
+                .OnSuccessCombineWith(publication => _methodologyService.GetSummariesByPublication(publication.Id))
+                .OnSuccess(async tuple =>
                 {
-                    var (viewModel, methodologies) = viewModelAndMethodologies;
-                    viewModel.Publication.Methodologies = methodologies;
-                    return viewModel;
+                    var (publication, methodologies) = tuple;
+                    var cachedRelease = await GetCachedRelease(publicationSlug, releaseSlug);
+
+                    if (cachedRelease.IsRight
+                        && cachedRelease.Right is not null
+                        && publication is not null)
+                    {
+                        var result = new Either<ActionResult, ReleaseViewModel>(new ReleaseViewModel(
+                            _mapper.Map<CachedReleaseViewModel>(cachedRelease.Right),
+                            _mapper.Map<PublicationViewModel>(publication)
+                        ));
+
+                        result.Right.Publication.Methodologies = methodologies;
+                        return result;
+                    }
+
+                    return new NotFoundResult();
```

Comment:
I'm not sure that publication can be null here? 
I'm not sure if you consider this any better/simpler but you might be interested to see it can be written combining another `OnSuccessCombineWith` and without  needing to wrap the `ReleaseViewModel` in an `Either`, and without the references to `.Right` as follows:

```
            return await _publicationService.Get(publicationSlug)
                .OnSuccessCombineWith(publication => _methodologyService.GetSummariesByPublication(publication.Id))
                .OnSuccessCombineWith(_ => GetCachedRelease(publicationSlug, releaseSlug))
                .OnSuccess<
                    ActionResult,
                    Tuple<PublicationViewModel, List<MethodologyVersionSummaryViewModel>, CachedReleaseViewModel?>,
                    ReleaseViewModel>(tuple =>
                {
                    var (publication, methodologies, cachedRelease) = tuple;
                    if (cachedRelease != null)
                    {
                        return new ReleaseViewModel(
                            _mapper.Map<CachedReleaseViewModel>(cachedRelease),
                            _mapper.Map<PublicationViewModel>(publication))
                        {
                            Publication =
                            {
                                Methodologies = methodologies
                            }
                        };
                    }

                    return new NotFoundResult();
                });
```

Reply:
I changed it slightly differently, but still follows the broad strokes of what you suggest.